app/jobs/create_transcoded_item_job.rb

```diff
@@ -0,0 +1,21 @@
+class CreateTranscodedItemJob < ApplicationJob
+  queue_as :within_30_seconds
+
+  def perform(audio_file, codec_conversion)
+    # Check that this transcoded_item was not created while this job was in the queue
+    return if TranscodedItem.find_by(audio_file:, codec_conversion:).present?
+
+    path = TranscodedItem.random_path
+    FileUtils.mkdir_p Pathname.new(path).parent
+    audio_file.convert(codec_conversion, path)
+
+    TranscodedItem.transaction do
+      # Check that the transcoded item was not created while we were executing
+      if TranscodedItem.find_by(audio_file:, codec_conversion:).present?
+        FileUtils.rm_f path
+      else
+        TranscodedItem.create(audio_file:, codec_conversion:, path:)
```

Comment:
I prefer to use `create!` in these situations. If we don't expect the create/update/destroy to fail and/or don't handle the boolean that is returned, we should raise an error so we know about the failure and can improve the code